### DISCUSSION TITLE
How do I get it to use more of my CPU?

### DISCUSSION THREAD

**Discussion by Njasa2k** [oobabooga/text-generation-webui#1050] [Q&A] [answered]
I have 8GB of VRAM and it is using 6GB of VRAM but barely any of the 64GB of regular RAM.
my settings: `call python server.py --auto-devices --chat --wbits 4 --groupsize 128 --gpu-memory 7 --pre_layer 30`
my inference speed is is about 0.6 tokens/second, I was hoping offloading it to the RAM would increase the speed.

**Comment by StefanIsMe** [comment]
Include the `--threads [#]` parameter. Change # to the number of cores available on your CPU.

**Reply by cparoli** [reply]
Unfortunately --threads doesn't change the speed.

**Comment by mcmonkey4eva** [comment, marked as answer]
Offloading to RAM *decreases* speed, not increase. It doesn't use your CPU, it just dumps GPU memory to system RAM and has to transfer it repeatedly while generating to function.

If you want to run in CPU mode, you will need to use `ggml-` models to run [llama.cpp mode](https://github.com/oobabooga/text-generation-webui/wiki/llama.cpp-models).

Be warned this can end up slower than your GPU depending on your specs.

**Reply by Njasa2k** [reply]
I have 64GB of RAM and 8GB of VRAM, is there any way to speed it up?

**Reply by cparoli** [reply]
Would someone have a more detailed documentation on how --pre_layer works?
